include/boost/math/special_functions/lanczos.hpp

```diff
@@ -3227,37 +3224,26 @@ struct undefined_lanczos : public boost::integral_constant<int, INT_MAX - 1> { }
 #endif
 #endif
 
-typedef mpl::list<
-   lanczos6m24, 
-/*   lanczos6, */
-   lanczos13m53, 
-/*   lanczos13, */
-   lanczos17m64, 
-   lanczos24m113, 
-   lanczos22,
-   lanczos32MP, 
-   lanczos35MP,
-   lanczos48MP,
-   lanczos49MP,
-   lanczos49MP_2,
-   lanczos58MP,
-   undefined_lanczos> lanczos_list;
-
 template <class Real, class Policy>
 struct lanczos
 {
-   typedef typename mpl::if_<
-      typename mpl::less_equal<
-         typename policies::precision<Real, Policy>::type,
-         boost::integral_constant<int, 0>
-      >::type,
-      boost::integral_constant<int, INT_MAX - 2>,
-      typename policies::precision<Real, Policy>::type
-   >::type target_precision;
-
-   typedef typename mpl::deref<typename mpl::find_if<
-      lanczos_list, 
-      mpl::less_equal<target_precision, mpl::_1> >::type>::type type;
+   static constexpr auto target_precision = policies::precision<Real, Policy>::type::value <= 0 ? (std::numeric_limits<int>::max)()-2 : 
+                                                                                                   policies::precision<Real, Policy>::type::value;
+
+   using type = typename std::conditional<(target_precision <= 24), lanczos6m24, 
```

Comment:
I'm being super-picky here, but `(target_precision <= lanczos6m24::value)` would be better than copy-and-pasting the values.

Reply:
Magic numbers have been removed in 07144e2.